algebird-core/src/main/scala/com/twitter/algebird/Ring.scala

```diff
@@ -161,6 +161,35 @@ class FromAlgebraRing[T](r: ARing[T]) extends Ring[T] {
   override def product(ts: TraversableOnce[T]): T = r.product(ts)
 }
 
+class UnsafeFromAlgebraRig[T](r: Rig[T]) extends Ring[T] {
```

Comment:
curious as to why this is extending Ring? can we not just extend Rig? Maybe document the rationale?